### PR TITLE
Workflows: Add triage labels to PRs

### LINKED ISSUE #428: Size labels
PRs aren't getting size labels anymore, since we moved the project out of the Hashicorp org. They were so useful IMO. Is it possible to add [Hashibot](https://github.com/apps/hashibot) back?

### PR DESCRIPTION
**Pull request by armsnyder**
## Description

closes #428 

New workflows for pull requests that automatically add `size/*` and other area-like labels to pull requests.

Tested in my fork.

## PR Checklist

- [x] Resource attributes match 1:1 the names and structure of the API resource in [the GitLab API documentation](https://docs.gitlab.com/ee/api/).
- [x] [Examples](/examples) are updated with:
    - A \*.tf file for the resource/s with at least one usage example
    - A \*.sh file for the resource/s with an import example (if applicable)
- [x] New resources have at minimum a basic test with three steps:
    - Create the resource
    - Update the attributes
    - Import the resource
- [x] No new `//lintignore` comments that came from copied code. Linter rules are meant to be enforced on new code.

### CODE REVIEW

File: .github/labeler-pr-triage.yml
Comment (timofurrer) [inline .github/labeler-pr-triage.yml:17]: Maybe we could be a little bit more specific here (which requires additional labels) :)

```suggestion
provider:
  - '.go-version'
  - 'main.go'
  - 'gitlab/provider_*.go'
  
  resource:
  - 'gitlab/resource_*.go'
  
  data-source:
  - 'gitlab/data_source_*.go'
```

**Review by timofurrer — APPROVED** [review]

File: .github/labeler-pr-triage.yml
Comment (armsnyder) [reply .github/labeler-pr-triage.yml:20]: I like it! Updated. I left the "provider" label globs as-is because I was thinking if it as "if you see this label, then the provider code changed". Thoughts?

Comment (timofurrer) [reply .github/labeler-pr-triage.yml:20]: Yes, sounds good :)

**Comment by timofurrer** [comment]
Are the labels created automatically if they don't exist, because the `resource` and `data-source` at least don't exist yet :)

**Comment by armsnyder** [comment]
@timofurrer 
> Are the labels created automatically if they don't exist, because the resource and data-source at least don't exist yet :)

Aah, no they are not. I'll create them now. Feel free to mess with colors.

**Comment by armsnyder** [comment]
@timofurrer I had a changed stashed, where I had the workflow label a PR as "needs-triage" when it is opened. I didn't commit it because I'm unclear on how important PR labels are. WDYT?

**Comment by timofurrer** [comment]
I like the idea. I'm sure we'll improve here and there anyways once it's merged and we noticed shortcomings :)

**Comment by armsnyder** [comment]
I can open a PR for an automatic `needs-triage` label on PRs, after #829 is merged, since it would conflict with that. My only reservations are...
1. Not wanting to generate more work for maintainers to go through and remove the label.
1. What would a `needs-triage` label on a PR mean exactly, and can document that in like a `MAINTAINERS.md` for processes like triage and release, so we agree on expectations?

**Pull request merged**
